src/lib/communication/DY.Lib.Communication.API.Predicates.fst

```diff
@@ -0,0 +1,138 @@
+module DY.Lib.Communication.API.Predicates
```

Comment:
Nit: rename to `DY.Lib.Communication.API.Invariants`? It seems more specific than `Predicates`